src/dsync/dsync.c

```diff
@@ -2501,7 +2501,9 @@ static void dsync_output_init_disjunction(
 static void dsync_output_free(struct dsync_output* output)
 {
     assert(list_empty(&output->linkage));
-    dsync_disjunction_free(output->disjunction);
+    if (output->disjunction != NULL) {
```

Comment:
How about putting this check into `dsync_disjunction_free()` instead of doing it here?  That would make it the same as `free()`, where it's just a no-op if it's passed a NULL pointer.

Reply:
make sense. I'll make the change